src/server/db_slice.h

```diff
@@ -524,6 +525,39 @@ class DbSlice {
   void CallChangeCallbacks(DbIndex id, const ChangeReq& cr) const;
 
  private:
+  struct CounterCondFlag {
+    void WaitUntilCounterIsZero() {
+      util::fb2::NoOpLock noop_lk_;
+      cond_var.wait(noop_lk_, [this]() { return mutating == 0; });
+    }
+
+    util::fb2::CondVarAny cond_var;
+    size_t mutating = 0;
+  };
+
+  class CounterConditionGuard {
```

Comment:
if you add equivalent `lock`, `unlock` methods to `LocalBlockingCounter` you will be able to use std::lock_guard instead of writing a custom class.